addons/html_builder/static/src/website_builder/plugins/translation_plugin.js

```diff
@@ -0,0 +1,310 @@
+import { Plugin } from "@html_editor/plugin";
+import { browser } from "@web/core/browser/browser";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { AttributeTranslateDialog } from "../translation_components/attributeTranslateDialog";
+import { SelectTranslateDialog } from "../translation_components/selectTranslateDialog";
+import {
+    localStorageNoDialogKey,
+    TranslatorInfoDialog,
+} from "../translation_components/translatorInfoDialog";
+
+export const translationSavableSelector =
+    "[data-oe-translation-source-sha], " +
+    "[data-oe-model][data-oe-id][data-oe-field], " +
+    '[placeholder*="data-oe-translation-source-sha="], ' +
+    '[title*="data-oe-translation-source-sha="], ' +
+    '[value*="data-oe-translation-source-sha="], ' +
+    '[alt*="data-oe-translation-source-sha="]';
+
+export function getTranslationEditableEls(rootEl) {
+    const translationSavableEls = rootEl.querySelectorAll(translationSavableSelector);
+    const textAreaEls = Array.from(rootEl.querySelectorAll("textarea")).find((el) =>
+        el.textContent.includes("data-oe-translation-source-sha")
+    );
+    return Array.from(translationSavableEls).concat(textAreaEls || []);
+}
+
+/**
+ *
+ * @param {HTMLElement} containerEl
+ * @returns {HTMLElement[]}
+ */
+function findOEditable(containerEl) {
+    const isOEditable = (node) => {
+        while (node) {
+            if (node.className && typeof node.className === "string") {
+                if (node.className.indexOf("o_not_editable") !== -1) {
+                    return false;
+                }
+                if (node.className.indexOf("o_editable") !== -1) {
+                    return true;
+                }
+            }
+            node = node.parentNode;
+        }
+        return false;
+    };
+    const allDescendants = containerEl.querySelectorAll("*");
+    return Array.from(allDescendants).filter(isOEditable);
+}
+
+export class TranslationPlugin extends Plugin {
+    static id = "translation";
+
+    resources = {
+        clean_for_save_handlers: this.cleanForSave.bind(this),
+        get_dirty_els_not_in_dom: this.getDirtyTranslations.bind(this),
+    };
+
+    setup() {
+        if (this.config.isTranslation) {
+            this.prepareTranslation();
+        }
+    }
+
+    prepareTranslation() {
+        this.websiteService = this.services.website;
+        this.notificationService = this.services.notification;
+        this.dialogService = this.services.dialog;
+        const editableEls = findOEditable(this.editable);
+        this.elToTranslationInfoMap = new Map();
+        const translatedAttrs = ["placeholder", "title", "alt", "value"];
```

Comment:
Note: for the moment, there is a problem when a form field has an `input` attribute. I will solve it in another PR; the problem is that interactions that should not be started in edit mode are somehow started and destroyed when entering translate mode.